cos-gpu-installer-docker/gpu_installer_url_lib.sh

```diff
@@ -53,14 +53,14 @@ default_installer_download_url() {
   if (( "${major_version}" < 390 )); then
     # Versions prior to 390 are downloaded from the upstream location.
     echo "https://us.download.nvidia.com/tesla/${driver_version}/NVIDIA-Linux-x86_64-${driver_version}.run"
-  elif (( "${major_version}" == 390 )); then
+  elif (( "${major_version}" == 390 )) && (( "${minor_version}" == 46 )); then
     # The naming format changed after version 390.
     echo "https://storage.googleapis.com/nvidia-drivers-${download_location}-public/TESLA/NVIDIA-Linux-x86_64-${driver_version}.run"
-  elif (( "${major_version}" >= 396 )) && (( "${minor_version}" >= 37 )); then
+  elif (( "${major_version}" == 396 )) && (( "${minor_version}" == 26 )); then
     # Apparently the naming format changed again starting since 396.37.
-    echo "https://storage.googleapis.com/nvidia-drivers-${download_location}-public/tesla/${driver_version}/NVIDIA-Linux-x86_64-${driver_version}.run"
-  else
     echo "https://storage.googleapis.com/nvidia-drivers-${download_location}-public/tesla/${driver_version}/NVIDIA-Linux-x86_64-${driver_version}-diagnostic.run"
+  else
+    echo "https://storage.googleapis.com/nvidia-drivers-${download_location}-public/tesla/${driver_version}/NVIDIA-Linux-x86_64-${driver_version}.run"
```

Comment:
Can you update the comments as well? Also, can you include comments that explain missing version ranges? Something like:

if (( "${major_version}" < 390 )); then
  # Versions prior to 390 ...
elif (( "${major_version}" == 390 ... )); then
  # 390.46 is the only 390 version
elif (( "${major_version}" == 396 ... )); then
  # There are no versions between 390.46 and 396.26
else
  # All versions over 396.26

Reply:
Updated the comments. There exists one version between 390.46 and 396.26 which is 390.116 but that one conforms to the naming format seen in the later versions.